k8s/plugin/src/resources/upgrade.rs

```diff
@@ -270,148 +372,156 @@ impl UpgradeResources {
+                                    .namespace
+                                    .as_ref()
+                                    .unwrap(),
+                            );
                         }
-                    }
-                    Err(e) => {
-                        println!("Failed in fetching cluster role binding  {:?}", e);
-                        std::process::exit(1);
-                    }
-                }
-
-                // delete cluster role
-                match duo.get_cluster_role().await {
-                    Ok(cr) => {
-                        if cr.iter().count() == 1 {
-                            match duo
-                                .cluster_role
+                        Actions::Delete => {
+                            match self
+                                .service
                                 .delete(
-                                    &upgrade_group(
-                                        &duo.release_name,
-                                        UPGRADE_OPERATOR_CLUSTER_ROLE,
-                                    ),
-                                    &dp,
+                                    &upgrade_group(&self.release_name, UPGRADE_OPERATOR_SERVICE),
+                                    &DeleteParams::default(),
                                 )
                                 .await
                             {
                                 Ok(_) => {
-                                    println!("cluster role deleted");
+                                    println!("service deleted");
                                 }
                                 Err(e) => {
-                                    println!("Failed in deleting cluster role {:?}", e);
+                                    println!("Failed in deleting service {:?}", e);
                                     tokio::time::sleep(Duration::from_secs(1)).await;
                                     std::process::exit(1);
                                 }
                             }
-                        } else {
-                            println!("cluster role does not exist!");
                         }
                     }
-                    Err(e) => {
-                        println!("Failed in fetching cluster role {:?}", e);
-                        std::process::exit(1);
-                    }
                 }
+            }
+
+            Err(e) => {
+                println!("Failed in fetching service {:?}", e);
+                std::process::exit(1);
+            }
+        }
+        Ok(())
+    }
 
+    /// Install the upgrade resources
+    pub async fn install(ns: &str) {
+        match UpgradeResources::new(ns).await {
+            Ok(uo) => {
+                // Create Service Account
+                uo.service_account_actions(ns, Actions::Create)
+                    .await
+                    .map_err(|err| println!("{:?}", err))
```

Comment:
we print the errors but ignore them, is this intended?

Reply:
yes, as we need to continue even if the resources are present.

Reply:
Should we specifically ignore AlreadyExists error then? Seems we'll just ignore any error.

Reply:
Resolved it. PTAL